File: backend/product/src/main/java/com/example/product/pojo/dto/ProductDto.java
package com.example.product.pojo.dto;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
@Data
public class ProductDto {
    private String productId;
    private String productName;
    private String categoryName;
    private Integer categoryId;
    private String productNote;
    private BigDecimal price;
    private BigDecimal memberPrice;
    private Integer stock;
    private String originPlace;
    private String shippingPlace;
    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    private LocalDateTime createTime;
    private String createdBy;
    //可以解决LocalDateTime类型无法被序列化存入redis的问题
    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    private LocalDateTime updateTime;
    private String updatedBy;
    private Boolean deletedFlag;
    private String reservedField1;
    private String reservedField2;

}

File: frontend/src/dto/ProductDto.ts
import React from "react";

export interface Product {
  productId: string; // 商品唯一标识（可选列）
  productName: string; // 商品名
  categoryName: string; // 商品分类ID（如需展示分类名，可以考虑加个 categoryName）
  categoryId: number;
  originPlace: string; // 产地
  shippingPlace: string; // 发货地
  price: number; // 价格
  memberPrice: number; // 会员价
  stock: number; // 库存
  updatedBy: string; // 更新人
  updateTime: string; // 更新时间，推荐用 ISO 字符串（如 "2025-07-25T12:00:00"）
  productNote: string;
}


export interface ProductAdding {
  productName: string;
  categoryId: number;
  originPlace: string;
  shippingPlace: string;
  price: string;
  memberPrice: string;
  stock: string;
  productNote: string;
}

export interface ProductEditing {
  productId: string; // 商品唯一标识（可选列）
  productName: string;
  categoryId: number;
  originPlace: string;
  shippingPlace: string;
  price: string;
  memberPrice: string;
  stock: string;
  productNote: string;
}

export interface ProductQuery {
  productName: string;
  categoryId: number;
  originPlace: string;
  shippingPlace: string;
}

export interface ProductPage {
  total:number;
  list:Product[];
}
